src/server/usage/README.md

```diff
@@ -0,0 +1,96 @@
+# Kibana Telemetry Service
+
+Telemetry allows Kibana features to have usage tracked in the wild. The general term "telemetry" refers to multiple things:
+
+1. Integrating with the telemetry service to express how to collect usage data (Collecting).
+2. Sending a payload of usage data up to Elastic's telemetry cluster, once per browser per day (Sending).
+3. Viewing usage data in the Kibana instance of the telemetry cluster (Viewing).
+
+You, the feature or plugin developer, mainly need to worry about the first meaning: collecting. To integrate with the telemetry services for usage collection of your feature, there are 2 steps:
+
+1. Create a usage collector using a factory function
+2. Register the usage collector with the Telemetry service
+
+NOTE: To a lesser extent, there's also a need to update the telemetry payload of Kibana stats and telemetry cluster field mappings to include your fields. This part is typically handled not by you, the developer, but different maintainers of the telemetry cluster. Usually, this step just means talk to the Platform team and have them approve your data model or added fields.
+
+## Creating a Usage Collector
+
+A usage collector object is an instance of a class called `UsageCollector`. A factory function on `server.usage.collectorSet` object allows you to create an instance of this class. All you need to provide is a `type` for organizing your fields, and a `fetch` method for returning your usage data.
+
+Example:
+
+```js
+const myCollector = server.usage.collectorSet.makeUsageCollector({
+  type: MY_USAGE_TYPE,
+  fetch: async callCluster => {
+
+    // query ES and get some data
+    // summarize the data into a model
+    // return the modeled object that includes whatever you want to track
+
+    return {
+      my_objects: {
+        total: SOME_NUMBER
+      }
+    };
+  },
+});
+```
```

Comment:
I think the background descriptions are fine but it is a little lengthy for users that just want to get started with telemetry. Perhaps you can present the creation and registration like:

```js
// create usage collector
const myCollector = server.usage.collectorSet.makeUsageCollector({
  type: MY_USAGE_TYPE,
  fetch: async callCluster => {

    // query ES and get some data
    // summarize the data into a model
    // return the modeled object that includes whatever you want to track

     return {
      my_objects: {
        total: SOME_NUMBER
      }
    };
  },
});

// register usage collector
server.usage.collectorSet.register(myCollector);
```